### ISSUE TITLE
plugin crashing homebridge since new version

### ISSUE THREAD

**Issue by pelletip** [iRayanKhan/homebridge-tuya#212]
**Checklist**
- [x] I have read the common issues wiki page
- [x] I have checked to make sure the plugin is up to date

**Describe the bug**
I have been using your plugin with my garage doors as well as a few smart switches that are using Tuya.  this has been very reliable until recently.  my config was working, everything was stable.  but since the new updates of of your plugin, I am getting errors in the logs.  I have pasted a few loops of the error messages, that are constantly cycling in my homebridge.

**To Reproduce**
using version 1.3.1 or 1.2.5 of homebridge and your plugin homebridge-tuya v1.4.0

**Expected behavior**
your plugin version 1.3.0 works well with homebridge v1.2.5 and 1.3.1 

**Screenshots**
I would be happy to provide anything to help you fix, as needed

**Environment (please complete the following information):**
 - OS: Raspberry pi
-  iOS version: 14.4
-  Homehubs: homebridge 1.3.1
 - Node Version: 14.16.0
 - Plugin Version: 1.4.0
-  Accessory Type: Garage Door

here is a copy of the config file info, for your plugin
 {
            "platform": "TuyaLan",
            "name": "TuyaLan",
            "devices": [
                {
                    "name": "Globe Switch 1",
                    "type": "Outlet",
                    "manufacturer": "Globe",
                    "model": "G1",
                    "id": "10487343xa029115045b",
                    "key": "71ca54cff3195f95"
                },
                {
                    "name": "Globe Switch 2",
                    "type": "Outlet",
                    "manufacturer": "Globe",
                    "model": "G1",
                    "id": "10487343b00291150d8d",
                    "key": "2391bf02b7x45bf8"
                },
                {
                    "name": "Globe Switch 3",
                    "type": "Outlet",
                    "manufacturer": "Globe",
                    "model": "G1",
                    "id": "104a734350029x14f351",
                    "key": "cx710d83bcd2b16c"
                },
                {
                    "name": "Garage Genevieve",
                    "type": "GarageDoor",
                    "manufacturer": "aabhvlh7e0jcqvbv",
                    "model": "Garage door",
                    "id": "05133168d8f15b85a89f",
                    "key": "5af70b640717b332"
                },
                {
                    "name": "Garage Pierre",
                    "type": "GarageDoor",
                    "manufacturer": "aabhvlh7e0jcqvbv",
                    "model": "Garage door",
                    "id": "03113168d8f15b85a0cd",
                    "key": "3f05z2e3ed7e9d8b"
                }
            ]
        }

**Additional context**

[25/02/2021, 17:43:21] [HB Supervisor] Restarting Homebridge...
[25/02/2021, 17:43:21] [HB Supervisor] Starting Homebridge with extra flags: -I -D
[25/02/2021, 17:43:21] [HB Supervisor] Started Homebridge v1.3.1 with PID: 1520
Initializing HAP-NodeJS v0.9.2...
[25/02/2021, 17:43:22] Loaded config.json with 0 accessories and 2 platforms.
[25/02/2021, 17:43:22] Loaded 5 cached accessories from cachedAccessories.
[25/02/2021, 17:43:22] ---
[25/02/2021, 17:43:23] Loaded plugin: homebridge-config-ui-x@4.39.1
[25/02/2021, 17:43:23] Registering platform 'homebridge-config-ui-x.config'
[25/02/2021, 17:43:23] ---
[25/02/2021, 17:43:23] Loaded plugin: homebridge-tuya@1.4.0
[25/02/2021, 17:43:23] Plugin 'homebridge-tuya' tried to register with an incorrect plugin identifier: 'homebridge-tuya-lan'. Please report this to the developer!
[25/02/2021, 17:43:23] Registering platform 'homebridge-tuya.TuyaLan'
[25/02/2021, 17:43:23] ---
[25/02/2021, 17:43:23] Loading 2 platforms...
[25/02/2021, 17:43:23] [Config] Initializing config platform...
[25/02/2021, 17:43:23] [Config] Running in Service Mode
[25/02/2021, 17:43:23] [TuyaLan] Initializing TuyaLan platform...
[25/02/2021, 17:43:23] [TuyaLan] Marked Globe Switch 3 unreachable by faulting Service.Globe Switch 3.On
[25/02/2021, 17:43:23] [TuyaLan] Marked Garage Pierre unreachable by faulting Service.Garage Pierre.Target Door State
[25/02/2021, 17:43:23] [TuyaLan] Marked Garage Genevieve unreachable by faulting Service.Garage Genevieve.Target Door State
[25/02/2021, 17:43:23] [TuyaLan] Marked Globe Switch 1 unreachable by faulting Service.Globe Switch 1.On
[25/02/2021, 17:43:23] [TuyaLan] Marked Globe Switch 2 unreachable by faulting Service.Globe Switch 2.On
[25/02/2021, 17:43:23] [TuyaLan] Starting discovery...
Preparing Advertiser for 'Homebridge 0D3B C19D' using bonjour-hap backend!
Setup Payload:
X-HM://0024BDYFNJILT
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ 111-11-111 │     
    └────────────┘     
                       
[TuyaDiscovery] Discovery started on port 6666.
[TuyaDiscovery] Discovery started on port 6667.
Starting to advertise 'Homebridge 0D3B C19D' using bonjour-hap backend!
[25/02/2021, 17:43:23] Homebridge v1.3.1 (Homebridge 0D3B) is running on port 51575.
[25/02/2021, 17:43:25] [TuyaLan] Discovered Globe Switch 3 (104a734350029x14f351) identified as Outlet (3.3)
[TuyaAccessory] Changing ping gap for Globe Switch 3 to 9s
[TuyaAccessory DEBUG] reconnect called for Globe Switch 3
[25/02/2021, 17:43:25] [TuyaLan] Connected to Globe Switch 3
[TuyaAccessory] Sending first query to Globe Switch 3 (3.3)
[TuyaAccessory] Heard back from Globe Switch 3 with command 10
[25/02/2021, 17:43:25] [TuyaLan] Ready to handle Globe Switch 3 (Outlet:3.3) with signature {"1":false,"9":0}
[25/02/2021, 17:43:26] [TuyaLan] Discovered Garage Pierre (03113168d8f15b85a0cd) identified as GarageDoor (3.3)
[TuyaAccessory] Changing ping gap for Garage Pierre to 9s
[TuyaAccessory DEBUG] reconnect called for Garage Pierre
[25/02/2021, 17:43:26] [TuyaLan] Connected to Garage Pierre
[TuyaAccessory] Sending first query to Garage Pierre (3.3)
[TuyaAccessory] Heard back from Garage Pierre with command 10
[25/02/2021, 17:43:26] [TuyaLan] Ready to handle Garage Pierre (GarageDoor:3.3) with signature {"1":false,"7":0,"101":false}
[TuyaAccessory] aabhvlh7e0jcqvbv GarageDoor _getTargetDoorState UNKNOWN STATE undefined
[25/02/2021, 17:43:26] [homebridge-tuya] This plugin generated a warning from the characteristic 'Target Door State': characteristic value expected number and received undefined. See https://git.io/JtMGR for more info.
[25/02/2021, 17:43:26] [homebridge-tuya] Error: 
    at TargetDoorState.Characteristic.characteristicWarning (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:2164:105)
    at TargetDoorState.Characteristic.validateUserInput (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:1996:16)
    at TargetDoorState.Characteristic.updateValue (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:1365:20)
    at GarageDoorAccessory._registerCharacteristics (/usr/local/lib/node_modules/homebridge-tuya/lib/GarageDoorAccessory.js:126:14)
    at TuyaAccessory.<anonymous> (/usr/local/lib/node_modules/homebridge-tuya/lib/BaseAccessory.js:22:18)
    at Object.onceWrapper (events.js:422:26)
    at TuyaAccessory.emit (events.js:315:20)
    at TuyaAccessory._change (/usr/local/lib/node_modules/homebridge-tuya/lib/TuyaAccessory.js:407:18)
    at TuyaAccessory._msgHandler_3_3 (/usr/local/lib/node_modules/homebridge-tuya/lib/TuyaAccessory.js:333:30)
    at /usr/local/lib/node_modules/homebridge-tuya/node_modules/async/dist/async.js:3201:13
[25/02/2021, 17:43:26] ReferenceError: dps is not defined
    at GarageDoorAccessory._getCurrentDoorState (/usr/local/lib/node_modules/homebridge-tuya/lib/GarageDoorAccessory.js:290:36)
    at GarageDoorAccessory._registerCharacteristics (/usr/local/lib/node_modules/homebridge-tuya/lib/GarageDoorAccessory.js:131:31)
    at TuyaAccessory.<anonymous> (/usr/local/lib/node_modules/homebridge-tuya/lib/BaseAccessory.js:22:18)
    at Object.onceWrapper (events.js:422:26)
    at TuyaAccessory.emit (events.js:315:20)
    at TuyaAccessory._change (/usr/local/lib/node_modules/homebridge-tuya/lib/TuyaAccessory.js:407:18)
    at TuyaAccessory._msgHandler_3_3 (/usr/local/lib/node_modules/homebridge-tuya/lib/TuyaAccessory.js:333:30)
    at /usr/local/lib/node_modules/homebridge-tuya/node_modules/async/dist/async.js:3201:13
    at Object.process (/usr/local/lib/node_modules/homebridge-tuya/node_modules/async/dist/async.js:1427:21)
    at /usr/local/lib/node_modules/homebridge-tuya/node_modules/async/dist/async.js:1285:23
[25/02/2021, 17:43:26] Got SIGTERM, shutting down Homebridge...
[25/02/2021, 17:43:27] [TuyaLan] Discovered Garage Genevieve (05133168d8f15b85a89f) identified as GarageDoor (3.3)
[TuyaAccessory] Changing ping gap for Garage Genevieve to 9s
[TuyaAccessory DEBUG] reconnect called for Garage Genevieve
[25/02/2021, 17:43:27] [TuyaLan] Connected to Garage Genevieve
[TuyaAccessory] Sending first query to Garage Genevieve (3.3)
[TuyaAccessory] Heard back from Garage Genevieve with command 10
[25/02/2021, 17:43:27] [TuyaLan] Ready to handle Garage Genevieve (GarageDoor:3.3) with signature {"1":false,"7":0,"101":false}
[TuyaAccessory] aabhvlh7e0jcqvbv GarageDoor _getTargetDoorState UNKNOWN STATE undefined
[25/02/2021, 17:43:27] [homebridge-tuya] This plugin generated a warning from the characteristic 'Target Door State': characteristic value expected number and received undefined. See https://git.io/JtMGR for more info.
[25/02/2021, 17:43:27] [homebridge-tuya] Error: 
    at TargetDoorState.Characteristic.characteristicWarning (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:2164:105)
    at TargetDoorState.Characteristic.validateUserInput (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:1996:16)
    at TargetDoorState.Characteristic.updateValue (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:1365:20)
    at GarageDoorAccessory._registerCharacteristics (/usr/local/lib/node_modules/homebridge-tuya/lib/GarageDoorAccessory.js:126:14)
    at TuyaAccessory.<anonymous> (/usr/local/lib/node_modules/homebridge-tuya/lib/BaseAccessory.js:22:18)
    at Object.onceWrapper (events.js:422:26)
    at TuyaAccessory.emit (events.js:315:20)
    at TuyaAccessory._change (/usr/local/lib/node_modules/homebridge-tuya/lib/TuyaAccessory.js:407:18)
    at TuyaAccessory._msgHandler_3_3 (/usr/local/lib/node_modules/homebridge-tuya/lib/TuyaAccessory.js:333:30)
    at /usr/local/lib/node_modules/homebridge-tuya/node_modules/async/dist/async.js:3201:13
[25/02/2021, 17:43:27] ReferenceError: dps is not defined
    at GarageDoorAccessory._getCurrentDoorState (/usr/local/lib/node_modules/homebridge-tuya/lib/GarageDoorAccessory.js:290:36)
    at GarageDoorAccessory._registerCharacteristics (/usr/local/lib/node_modules/homebridge-tuya/lib/GarageDoorAccessory.js:131:31)
    at TuyaAccessory.<anonymous> (/usr/local/lib/node_modules/homebridge-tuya/lib/BaseAccessory.js:22:18)
    at Object.onceWrapper (events.js:422:26)
    at TuyaAccessory.emit (events.js:315:20)
    at TuyaAccessory._change (/usr/local/lib/node_modules/homebridge-tuya/lib/TuyaAccessory.js:407:18)
    at TuyaAccessory._msgHandler_3_3 (/usr/local/lib/node_modules/homebridge-tuya/lib/TuyaAccessory.js:333:30)
    at /usr/local/lib/node_modules/homebridge-tuya/node_modules/async/dist/async.js:3201:13
    at Object.process (/usr/local/lib/node_modules/homebridge-tuya/node_modules/async/dist/async.js:1427:21)
    at /usr/local/lib/node_modules/homebridge-tuya/node_modules/async/dist/async.js:1285:23
[25/02/2021, 17:43:28] [TuyaLan] Discovered Globe Switch 1 (10487343xa029115045b) identified as Outlet (3.3)
[TuyaAccessory] Changing ping gap for Globe Switch 1 to 9s
[TuyaAccessory DEBUG] reconnect called for Globe Switch 1
[25/02/2021, 17:43:28] [TuyaLan] Connected to Globe Switch 1
[TuyaAccessory] Sending first query to Globe Switch 1 (3.3)
[TuyaAccessory] Heard back from Globe Switch 1 with command 10
[25/02/2021, 17:43:28] [TuyaLan] Ready to handle Globe Switch 1 (Outlet:3.3) with signature {"1":true,"9":0}
[25/02/2021, 17:43:28] [TuyaLan] Discovered Globe Switch 2 (10487343b00291150d8d) identified as Outlet (3.3)
[TuyaAccessory] Changing ping gap for Globe Switch 2 to 9s
[TuyaAccessory DEBUG] reconnect called for Globe Switch 2
[TuyaAccessory] Discovery ended.
[25/02/2021, 17:43:28] [TuyaLan] Connected to Globe Switch 2
[TuyaAccessory] Sending first query to Globe Switch 2 (3.3)
[TuyaAccessory] Heard back from Globe Switch 2 with command 10
[25/02/2021, 17:43:28] [TuyaLan] Ready to handle Globe Switch 2 (Outlet:3.3) with signature {"1":false,"9":0}
[25/02/2021, 17:43:31] [HB Supervisor] Homebridge Process Ended. Code: 143, Signal: null
[25/02/2021, 17:43:36] [HB Supervisor] Restarting Homebridge...
[25/02/2021, 17:43:36] [HB Supervisor] Starting Homebridge with extra flags: -I -D
[25/02/2021, 17:43:36] [HB Supervisor] Started Homebridge v1.3.1 with PID: 1547
Initializing HAP-NodeJS v0.9.2...
[25/02/2021, 17:43:37] Loaded config.json with 0 accessories and 2 platforms.
[25/02/2021, 17:43:37] Loaded 5 cached accessories from cachedAccessories.
[25/02/2021, 17:43:37] ---
[25/02/2021, 17:43:38] Loaded plugin: homebridge-config-ui-x@4.39.1
[25/02/2021, 17:43:38] Registering platform 'homebridge-config-ui-x.config'
[25/02/2021, 17:43:38] ---
[25/02/2021, 17:43:38] Loaded plugin: homebridge-tuya@1.4.0
[25/02/2021, 17:43:38] Plugin 'homebridge-tuya' tried to register with an incorrect plugin identifier: 'homebridge-tuya-lan'. Please report this to the developer!
[25/02/2021, 17:43:38] Registering platform 'homebridge-tuya.TuyaLan'
[25/02/2021, 17:43:38] ---
[25/02/2021, 17:43:38] Loading 2 platforms...
[25/02/2021, 17:43:38] [Config] Initializing config platform...
[25/02/2021, 17:43:38] [Config] Running in Service Mode
[25/02/2021, 17:43:38] [TuyaLan] Initializing TuyaLan platform...
[25/02/2021, 17:43:38] [TuyaLan] Marked Globe Switch 3 unreachable by faulting Service.Globe Switch 3.On
[25/02/2021, 17:43:38] [TuyaLan] Marked Garage Pierre unreachable by faulting Service.Garage Pierre.Target Door State
[25/02/2021, 17:43:38] [TuyaLan] Marked Garage Genevieve unreachable by faulting Service.Garage Genevieve.Target Door State
[25/02/2021, 17:43:38] [TuyaLan] Marked Globe Switch 1 unreachable by faulting Service.Globe Switch 1.On
[25/02/2021, 17:43:38] [TuyaLan] Marked Globe Switch 2 unreachable by faulting Service.Globe Switch 2.On
[25/02/2021, 17:43:38] [TuyaLan] Starting discovery...
Preparing Advertiser for 'Homebridge 0D3B C19D' using bonjour-hap backend!
Setup Payload:
X-HM://0024BDYFNJILT
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ 111-11-111 │     
    └────────────┘     
                       
[TuyaDiscovery] Discovery started on port 6666.
[TuyaDiscovery] Discovery started on port 6667.
Starting to advertise 'Homebridge 0D3B C19D' using bonjour-hap backend!
[25/02/2021, 17:43:38] Homebridge v1.3.1 (Homebridge 0D3B) is running on port 51575.
[25/02/2021, 17:43:40] [TuyaLan] Discovered Globe Switch 3 (104a734350029x14f351) identified as Outlet (3.3)
[TuyaAccessory] Changing ping gap for Globe Switch 3 to 9s
[TuyaAccessory DEBUG] reconnect called for Globe Switch 3
[25/02/2021, 17:43:40] [TuyaLan] Connected to Globe Switch 3
[TuyaAccessory] Sending first query to Globe Switch 3 (3.3)
[TuyaAccessory] Heard back from Globe Switch 3 with command 10
[25/02/2021, 17:43:40] [TuyaLan] Ready to handle Globe Switch 3 (Outlet:3.3) with signature {"1":false,"9":0}
[25/02/2021, 17:43:41] [TuyaLan] Discovered Garage Pierre (03113168d8f15b85a0cd) identified as GarageDoor (3.3)
[TuyaAccessory] Changing ping gap for Garage Pierre to 9s
[TuyaAccessory DEBUG] reconnect called for Garage Pierre
[25/02/2021, 17:43:41] [TuyaLan] Connected to Garage Pierre
[TuyaAccessory] Sending first query to Garage Pierre (3.3)
[TuyaAccessory] Heard back from Garage Pierre with command 10
[25/02/2021, 17:43:41] [TuyaLan] Ready to handle Garage Pierre (GarageDoor:3.3) with signature {"1":false,"7":0,"101":false}
[TuyaAccessory] aabhvlh7e0jcqvbv GarageDoor _getTargetDoorState UNKNOWN STATE undefined
[25/02/2021, 17:43:41] [homebridge-tuya] This plugin generated a warning from the characteristic 'Target Door State': characteristic value expected number and received undefined. See https://git.io/JtMGR for more info.
[25/02/2021, 17:43:41] [homebridge-tuya] Error: 
    at TargetDoorState.Characteristic.characteristicWarning (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:2164:105)
    at TargetDoorState.Characteristic.validateUserInput (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:1996:16)
    at TargetDoorState.Characteristic.updateValue (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:1365:20)
    at GarageDoorAccessory._registerCharacteristics (/usr/local/lib/node_modules/homebridge-tuya/lib/GarageDoorAccessory.js:126:14)
    at TuyaAccessory.<anonymous> (/usr/local/lib/node_modules/homebridge-tuya/lib/BaseAccessory.js:22:18)
    at Object.onceWrapper (events.js:422:26)
    at TuyaAccessory.emit (events.js:315:20)
    at TuyaAccessory._change (/usr/local/lib/node_modules/homebridge-tuya/lib/TuyaAccessory.js:407:18)
    at TuyaAccessory._msgHandler_3_3 (/usr/local/lib/node_modules/homebridge-tuya/lib/TuyaAccessory.js:333:30)
    at /usr/local/lib/node_modules/homebridge-tuya/node_modules/async/dist/async.js:3201:13
[25/02/2021, 17:43:41] ReferenceError: dps is not defined
    at GarageDoorAccessory._getCurrentDoorState (/usr/local/lib/node_modules/homebridge-tuya/lib/GarageDoorAccessory.js:290:36)
    at GarageDoorAccessory._registerCharacteristics (/usr/local/lib/node_modules/homebridge-tuya/lib/GarageDoorAccessory.js:131:31)
    at TuyaAccessory.<anonymous> (/usr/local/lib/node_modules/homebridge-tuya/lib/BaseAccessory.js:22:18)
    at Object.onceWrapper (events.js:422:26)
    at TuyaAccessory.emit (events.js:315:20)
    at TuyaAccessory._change (/usr/local/lib/node_modules/homebridge-tuya/lib/TuyaAccessory.js:407:18)
    at TuyaAccessory._msgHandler_3_3 (/usr/local/lib/node_modules/homebridge-tuya/lib/TuyaAccessory.js:333:30)
    at /usr/local/lib/node_modules/homebridge-tuya/node_modules/async/dist/async.js:3201:13
    at Object.process (/usr/local/lib/node_modules/homebridge-tuya/node_modules/async/dist/async.js:1427:21)
    at /usr/local/lib/node_modules/homebridge-tuya/node_modules/async/dist/async.js:1285:23
[25/02/2021, 17:43:41] Got SIGTERM, shutting down Homebridge...
[25/02/2021, 17:43:42] [TuyaLan] Discovered Garage Genevieve (05133168d8f15b85a89f) identified as GarageDoor (3.3)
[TuyaAccessory] Changing ping gap for Garage Genevieve to 9s
[TuyaAccessory DEBUG] reconnect called for Garage Genevieve
[25/02/2021, 17:43:42] [TuyaLan] Connected to Garage Genevieve
[TuyaAccessory] Sending first query to Garage Genevieve (3.3)
[TuyaAccessory] Heard back from Garage Genevieve with command 10
[25/02/2021, 17:43:42] [TuyaLan] Ready to handle Garage Genevieve (GarageDoor:3.3) with signature {"1":false,"7":0,"101":false}
[TuyaAccessory] aabhvlh7e0jcqvbv GarageDoor _getTargetDoorState UNKNOWN STATE undefined
[25/02/2021, 17:43:42] [homebridge-tuya] This plugin generated a warning from the characteristic 'Target Door State': characteristic value expected number and received undefined. See https://git.io/JtMGR for more info.
[25/02/2021, 17:43:42] [homebridge-tuya] Error: 
    at TargetDoorState.Characteristic.characteristicWarning (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:2164:105)
    at TargetDoorState.Characteristic.validateUserInput (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:1996:16)
    at TargetDoorState.Characteristic.updateValue (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:1365:20)
    at GarageDoorAccessory._registerCharacteristics (/usr/local/lib/node_modules/homebridge-tuya/lib/GarageDoorAccessory.js:126:14)
    at TuyaAccessory.<anonymous> (/usr/local/lib/node_modules/homebridge-tuya/lib/BaseAccessory.js:22:18)
    at Object.onceWrapper (events.js:422:26)
    at TuyaAccessory.emit (events.js:315:20)
    at TuyaAccessory._change (/usr/local/lib/node_modules/homebridge-tuya/lib/TuyaAccessory.js:407:18)
    at TuyaAccessory._msgHandler_3_3 (/usr/local/lib/node_modules/homebridge-tuya/lib/TuyaAccessory.js:333:30)
    at /usr/local/lib/node_modules/homebridge-tuya/node_modules/async/dist/async.js:3201:13
[25/02/2021, 17:43:42] ReferenceError: dps is not defined
    at GarageDoorAccessory._getCurrentDoorState (/usr/local/lib/node_modules/homebridge-tuya/lib/GarageDoorAccessory.js:290:36)
    at GarageDoorAccessory._registerCharacteristics (/usr/local/lib/node_modules/homebridge-tuya/lib/GarageDoorAccessory.js:131:31)
    at TuyaAccessory.<anonymous> (/usr/local/lib/node_modules/homebridge-tuya/lib/BaseAccessory.js:22:18)
    at Object.onceWrapper (events.js:422:26)
    at TuyaAccessory.emit (events.js:315:20)
    at TuyaAccessory._change (/usr/local/lib/node_modules/homebridge-tuya/lib/TuyaAccessory.js:407:18)
    at TuyaAccessory._msgHandler_3_3 (/usr/local/lib/node_modules/homebridge-tuya/lib/TuyaAccessory.js:333:30)
    at /usr/local/lib/node_modules/homebridge-tuya/node_modules/async/dist/async.js:3201:13
    at Object.process (/usr/local/lib/node_modules/homebridge-tuya/node_modules/async/dist/async.js:1427:21)
    at /usr/local/lib/node_modules/homebridge-tuya/node_modules/async/dist/async.js:1285:23

....

**Comment by galzil** [comment]
The new [Child Bridge](https://github.com/homebridge/homebridge/releases/tag/v1.3.0) option could help here

**Comment by pelletip** [comment]
Hello, thanks for the reply
I did try and the child bridge helps isolate the rest of the processes from your plugin crashes.  But it does not solve the problem of the plugin crashing.

The 5 devices registered with your plugin are no longer responding, when I roll out 1.4 and up.

The log shows a message about an unknown state for the garage door and a warning about unexpected values for the TargetDoorState characteristic
[TuyaAccessory] aabhvlh7e0jcqvbv GarageDoor _getTargetDoorState UNKNOWN STATE undefined
[25/02/2021, 17:43:26] [homebridge-tuya] This plugin generated a warning from the characteristic 'Target Door State': characteristic value expected number and received undefined. See https://git.io/JtMGR for more info.

Then the plugin crashes, with a dsp not define error

[25/02/2021, 17:43:26] [homebridge-tuya] Error:
at TargetDoorState.Characteristic.characteristicWarning (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:2164:105)
at TargetDoorState.Characteristic.validateUserInput (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:1996:16)
at TargetDoorState.Characteristic.updateValue (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:1365:20)
at GarageDoorAccessory._registerCharacteristics (/usr/local/lib/node_modules/homebridge-tuya/lib/GarageDoorAccessory.js:126:14)
at TuyaAccessory. (/usr/local/lib/node_modules/homebridge-tuya/lib/BaseAccessory.js:22:18)
at Object.onceWrapper (events.js:422:26)
at TuyaAccessory.emit (events.js:315:20)
at TuyaAccessory._change (/usr/local/lib/node_modules/homebridge-tuya/lib/TuyaAccessory.js:407:18)
at TuyaAccessory._msgHandler_3_3 (/usr/local/lib/node_modules/homebridge-tuya/lib/TuyaAccessory.js:333:30)
at /usr/local/lib/node_modules/homebridge-tuya/node_modules/async/dist/async.js:3201:13
[25/02/2021, 17:43:26] ReferenceError: dps is not defined

When I rollback to 1.3.0, the problem goes away.

Thanks for your support,
Your plugin is amazing, I hope I can assist you in solving the problem

Best regards,
Pierre

**Comment by pelletip** [comment]
hello again, I did a little more testing with your latest version, to try to isolate the problem

I think the bug was introduced when new logic was recently inserted to support the Kogan brand.

My garage door is not Kogan, but if I set "manufacturer": "Kogan" in my config, the plugin still does not work, but it not longer crash.

when I have Kogan set as manufacturer, the garage door portion of the log provides door states warnings:
[TuyaAccessory DEBUG] decrementing this._connectionAttempts, currently 2
[TuyaAccessory DEBUG] decrementing this._connectionAttempts, currently 1
[TuyaAccessory] Kogan GarageDoor _getCurrentDoorState UNKNOWN STATUS undefined
[28/02/2021, 10:54:35] [homebridge-tuya] This plugin generated a warning from the characteristic 'Current Door State': characteristic value expected number and received undefined. See https://git.io/JtMGR for more info.
[TuyaAccessory] Kogan GarageDoor _getTargetDoorState UNKNOWN STATE undefined
[28/02/2021, 10:54:35] [homebridge-tuya] This plugin generated a warning from the characteristic 'Target Door State': characteristic value expected number and received undefined. See https://git.io/JtMGR for more info.
[TuyaAccessory] Kogan GarageDoor _getCurrentDoorState UNKNOWN STATUS undefined
[28/02/2021, 10:54:35] [homebridge-tuya] This plugin generated a warning from the characteristic 'Current Door State': characteristic value expected number and received undefined. See https://git.io/JtMGR for more info.
[TuyaAccessory] Kogan GarageDoor _getTargetDoorState UNKNOWN STATE undefined
[28/02/2021, 10:54:35] [homebridge-tuya] This plugin generated a warning from the characteristic 'Target Door State': characteristic value expected number and received undefined. See https://git.io/JtMGR for more info.
